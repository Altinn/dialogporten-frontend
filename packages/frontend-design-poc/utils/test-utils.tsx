import { QueryClient, QueryClientProvider } from "react-query";
import { ReactElement } from "react";
import { MemoryRouter } from "react-router-dom";
import { render, RenderOptions } from "@testing-library/react";

interface IExtendedRenderOptions extends RenderOptions {
  initialEntries?: string[];
}

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

export const customRender = (
  ui: ReactElement,
  options?: Omit<IExtendedRenderOptions, "wrapper">,
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <MemoryRouter initialEntries={options?.initialEntries ?? ["/"]}>
          {children}
        </MemoryRouter>
      </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: Wrapper, ...options });
};
