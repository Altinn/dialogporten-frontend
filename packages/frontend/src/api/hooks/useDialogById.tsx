import type { AttachmentLinkProps, AvatarProps, SeenByLogProps } from '@altinn/altinn-components';
import { useQuery, useQueryClient } from '@tanstack/react-query';
import {
  type Actor,
  ActorType,
  type AttachmentFieldsFragment,
  type DialogByIdFieldsFragment,
  type DialogStatus,
  type GetDialogByIdQuery,
  type OrganizationFieldsFragment,
  type PartyFieldsFragment,
  SystemLabel,
} from 'bff-types-generated';
import { AttachmentUrlConsumer } from 'bff-types-generated';
import { t } from 'i18next';
import type { DialogActionProps } from '../../components';
import { QUERY_KEYS } from '../../constants/queryKeys.ts';
import { type ValueType, getPreferredPropertyByLocale } from '../../i18n/property.ts';
import { useFormat } from '../../i18n/useDateFnsLocale.tsx';
import type { FormatFunction } from '../../i18n/useDateFnsLocale.tsx';
import { useOrganizations } from '../../pages/Inbox/useOrganizations.ts';
import { toTitleCase } from '../../profile';
import { graphQLSDK } from '../queries.ts';
import { type ActivityLogEntry, getActivityHistory } from '../utils/activities.tsx';
import { getSeenByLabel } from '../utils/dialog.ts';
import { type OrganizationOutput, getOrganization } from '../utils/organizations.ts';
import { type TimelineSegmentWithTransmissions, getTransmissions } from '../utils/transmissions.ts';
import { getViewTypes } from '../utils/viewType.ts';
import type { InboxViewType } from './useDialogs.tsx';
import { type SelectedPartyType, useParties } from './useParties.ts';

export enum EmbeddableMediaType {
  markdown = 'application/vnd.dialogporten.frontchannelembed-url;type=text/markdown',
  html = 'application/vnd.dialogporten.frontchannelembed-url;type=text/html',
}

export interface EmbeddedContent {
  /* URL to content provided by service owner */
  url: string;
  /* Media type of content */
  mediaType: EmbeddableMediaType;
}

export interface DialogByIdDetails {
  /* id of dialog */
  id: string;
  /* summary of dialog by locale,sorted by preference */
  summary: string;
  /* sender of dialog, fallbacks to dialog's service owner */
  sender: AvatarProps;
  /* recipient of dialog, fallbacks to current end-user */
  receiver: AvatarProps;
  /* title of dialog by locale,sorted by preference */
  title: string;
  /* actions available for dialog  */
  guiActions: DialogActionProps[];
  /* additional info of dialog by locale, sorted by preference - can be markdown / HTML */
  additionalInfo: { value: string; mediaType: string } | undefined;
  /* attachments for dialog, same of each attachment can be represented differently through list of URLs  */
  attachments: AttachmentLinkProps[];
  /* Used for guiActions and front channel embeds */
  dialogToken: string;
  /* main content reference for dialog, used to dynamically embed content in the frontend from an external URL. */
  mainContentReference?: EmbeddedContent;
  /* all activities for dialog, including transmissions */
  activityHistory: ActivityLogEntry[];
  /* last updated timestamp */
  updatedAt: string;
  /* created timestamp */
  createdAt: string;
  /* list of system labels for the dialog */
  label: SystemLabel[];
  /* all transmissions for dialog, grouped by relationship */
  transmissions: TimelineSegmentWithTransmissions[];
  /* a map of all content references for all content reference for transmission, used to dynamically embed content in the frontend from an external URL. */
  contentReferenceForTransmissions: Record<string, EmbeddedContent>;
  /* dialog status */
  status: DialogStatus;
  /* all actions (including end-user) that have seen the dialog since last update */
  seenByLog: SeenByLogProps;
  /* due date for dialog: This is the last date when the dialog is expected to be completed. */
  dueAt?: string;
  /* view type of dialog, used for grouping in inbox */
  viewType: InboxViewType;
}

interface UseDialogByIdOutput {
  isSuccess: boolean;
  isError: boolean;
  isLoading: boolean;
  dialog?: DialogByIdDetails;
  isAuthLevelTooLow?: boolean;
}
export const getDialogsById = (id: string): Promise<GetDialogByIdQuery> =>
  graphQLSDK.getDialogById({
    id,
  });

export const getAttachmentLinks = (attachments: AttachmentFieldsFragment[]): AttachmentLinkProps[] => {
  return attachments
    .filter((a) => a.urls.filter((url) => url.consumerType === AttachmentUrlConsumer.Gui).length > 0)
    .flatMap((attachment) =>
      attachment.urls
        .filter((url) => url.url !== 'urn:dialogporten:unauthorized')
        .map((url) => ({
          label: getPreferredPropertyByLocale(attachment.displayName)?.value || url.url,
          href: url.url,
        })),
    );
};

const getMainContentReference = (
  args: { value: ValueType; mediaType: string } | undefined | null,
  isAuthorized: boolean,
): EmbeddedContent | undefined => {
  if (!isAuthorized || typeof args === 'undefined' || args === null) return undefined;

  const { value, mediaType } = args;
  const content = getPreferredPropertyByLocale(value);
  const isValidMediaType = Object.values(EmbeddableMediaType).includes(mediaType as EmbeddableMediaType);

  if (!content || !isValidMediaType) return undefined;

  return {
    url: content.value,
    mediaType: mediaType as EmbeddableMediaType,
  };
};

export const getActorProps = (actor: Actor, serviceOwner?: OrganizationOutput) => {
  const isCompany =
    actor.actorType === ActorType.ServiceOwner || (actor.actorId ?? '').includes('urn:altinn:organization:');
  const type: AvatarProps['type'] = isCompany ? 'company' : 'person';
  const hasSenderName = (actor.actorName?.length ?? 0) > 0;
  const senderName = hasSenderName ? toTitleCase(actor.actorName) : (serviceOwner?.name ?? '');
  const senderLogo = isCompany ? serviceOwner?.logo : undefined;
  return {
    name: senderName,
    type,
    imageUrl: senderLogo,
    imageUrlAlt: t('dialog.imageAltURL', { companyName: senderName }),
  };
};

export function mapDialogToToInboxItem(
  item: DialogByIdFieldsFragment | null | undefined,
  parties: PartyFieldsFragment[],
  organizations: OrganizationFieldsFragment[],
  format: FormatFunction,
  selectedProfile: SelectedPartyType,
): DialogByIdDetails | undefined {
  if (!item) {
    return undefined;
  }
  const clockPrefix = t('word.clock_prefix');
  const formatString = `do MMMM yyyy ${clockPrefix ? `'${clockPrefix}' ` : ''}HH.mm`;
  const titleObj = item?.content?.title?.value;
  const additionalInfoObj = item?.content?.additionalInfo?.value;
  const summaryObj = item?.content?.summary?.value;
  const mainContentReference = item?.content?.mainContentReference;
  const endUserParty = parties?.find((party) => party.isCurrentEndUser);
  const dialogRecipientParty = parties?.find((party) => party.party === item.party);
  const actualRecipientParty = dialogRecipientParty ?? endUserParty;
  const serviceOwner = getOrganization(organizations || [], item.org, 'nb');
  const senderName = item.content.senderName?.value;
  const { seenByLabel } = getSeenByLabel(item.seenSinceLastUpdate, t);

  return {
    id: item.id,
    title: getPreferredPropertyByLocale(titleObj)?.value ?? '',
    status: item.status,
    summary: getPreferredPropertyByLocale(summaryObj)?.value ?? '',
    sender: {
      name: getPreferredPropertyByLocale(senderName)?.value || serviceOwner?.name || '',
      type: 'company',
      imageUrl: serviceOwner?.logo,
      imageUrlAlt: t('dialog.imageAltURL', { companyName: getPreferredPropertyByLocale(senderName)?.value }),
    },
    receiver: {
      name: actualRecipientParty?.name ?? '',
      type: actualRecipientParty?.partyType === 'Organization' ? 'company' : 'person',
    },
    additionalInfo: {
      value: getPreferredPropertyByLocale(additionalInfoObj)?.value ?? '',
      mediaType: item.content?.additionalInfo?.mediaType ?? '',
    },
    guiActions: item.guiActions.map((guiAction) => ({
      id: guiAction.id,
      url: guiAction.url,
      hidden:
        !guiAction.isAuthorized ||
        (guiAction.isDeleteDialogAction && !item.endUserContext?.systemLabels.includes(SystemLabel.Bin)),
      priority: guiAction.priority,
      httpMethod: guiAction.httpMethod,
      title: getPreferredPropertyByLocale(guiAction.title)?.value ?? '',
      prompt: getPreferredPropertyByLocale(guiAction.prompt)?.value,
      isDeleteAction: guiAction.isDeleteDialogAction,
      disabled: !guiAction.isAuthorized,
    })),
    attachments: getAttachmentLinks(item.attachments),
    mainContentReference: getMainContentReference(mainContentReference, true),
    contentReferenceForTransmissions: item.transmissions.reduce(
      (acc, transmission) => {
        const reference = getMainContentReference(transmission.content?.contentReference, transmission.isAuthorized);
        if (reference) {
          acc[transmission.id] = reference;
        }
        return acc;
      },
      {} as Record<string, EmbeddedContent>,
    ),
    dialogToken: item.dialogToken!,
    seenByLog: {
      collapsible: true,
      title: seenByLabel,
      endUserLabel: t('word.you'),
      items: item.seenSinceLastUpdate.map((seen) => ({
        id: seen.id,
        isEndUser: seen.isCurrentEndUser,
        name: (seen?.isCurrentEndUser ? (endUserParty?.name ?? '') : toTitleCase(seen.seenBy?.actorName ?? '')) || '',
        seenAt: seen.seenAt,
        seenAtLabel: format(seen.seenAt, formatString),
        type: 'person',
      })),
    },
    activityHistory: getActivityHistory({
      activities: item.activities,
      transmissions: item.transmissions,
      format,
      serviceOwner,
      selectedProfile,
    }),
    transmissions: getTransmissions({
      transmissions: item.transmissions,
      format,
      activities: item.activities,
      serviceOwner,
      selectedProfile,
    }),
    createdAt: item.createdAt,
    updatedAt: item.updatedAt,
    label: item.endUserContext?.systemLabels,
    viewType: getViewTypes({ status: item.status, systemLabel: item.endUserContext?.systemLabels }, true)?.[0],
    dueAt: item.dueAt,
  };
}

export const useDialogById = (parties: PartyFieldsFragment[], id?: string): UseDialogByIdOutput => {
  const queryClient = useQueryClient();
  const format = useFormat();
  const { organizations, isLoading: isOrganizationsLoading } = useOrganizations();
  const { selectedProfile } = useParties();
  const partyURIs = parties.map((party) => party.party);
  const { data, isSuccess, isLoading, isError } = useQuery<GetDialogByIdQuery>({
    queryKey: [QUERY_KEYS.DIALOG_BY_ID, id, organizations],
    staleTime: 1000 * 60 * 30,
    refetchInterval: 1000 * 60 * 10,
    refetchOnWindowFocus: false,
    retry: 3,
    queryFn: () =>
      getDialogsById(id!).then((data) => {
        if (data?.dialogById.dialog) {
          void queryClient.invalidateQueries({ queryKey: [QUERY_KEYS.COUNT_DIALOGS] });
        }
        return data;
      }),
    enabled: typeof id !== 'undefined' && partyURIs.length > 0,
  });

  if (isOrganizationsLoading) {
    return { isLoading: true, isError: false, isSuccess: false };
  }

  return {
    isLoading,
    isSuccess,
    dialog: mapDialogToToInboxItem(data?.dialogById.dialog, parties, organizations, format, selectedProfile),
    isError,
    isAuthLevelTooLow:
      data?.dialogById?.errors?.some((error) => error.__typename === 'DialogByIdForbiddenAuthLevelTooLow') ?? false,
  };
};
