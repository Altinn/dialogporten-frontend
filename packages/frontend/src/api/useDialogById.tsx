import { useQuery } from '@tanstack/react-query';
import type {
  AttachmentFieldsFragment,
  DialogActivityFragment,
  DialogByIdFieldsFragment,
  GetDialogByIdQuery,
  PartyFieldsFragment,
} from 'bff-types-generated';
import type { GuiActionButtonProps, InboxItemMetaField } from '../components';
import { QUERY_KEYS } from '../constants/queryKeys.ts';
import { i18n } from '../i18n/config.ts';
import { getOrganisation } from './organisations.ts';
import { graphQLSDK } from './queries.ts';

export interface Participant {
  name: string;
  isCompany: boolean;
  imageURL?: string;
}

interface MainContentReference {
  url: string;
  mediaType: 'markdown' | 'html' | 'unknown';
}

export interface DialogActivity {
  id: string;
  type: DialogActivityFragment['type'];
  createdAt: string;
  description: string;
  performedBy: DialogActivityFragment['performedBy'];
}
export interface DialogByIdDetails {
  summary: string;
  sender: Participant;
  receiver: Participant;
  title: string;
  metaFields: InboxItemMetaField[];
  guiActions: GuiActionButtonProps[];
  additionalInfo: string | React.ReactNode;
  attachments: AttachmentFieldsFragment[];
  dialogToken: string;
  mainContentReference?: MainContentReference;
  activities: DialogActivity[];
  updatedAt: string;
  createdAt: string;
}

interface UseDialogByIdOutput {
  isSuccess: boolean;
  isLoading: boolean;
  dialog?: DialogByIdDetails;
}
export const getDialogsById = (id: string): Promise<GetDialogByIdQuery> =>
  graphQLSDK.getDialogById({
    id,
  });

type ValueType =
  | Array<{
      __typename?: 'Localization';
      value: string;
      languageCode: string;
    }>
  | null
  | undefined;

export const getPropertyByCultureCode = (value: ValueType): string => {
  const defaultCultureCodes = ['nb'];
  if (value) {
    return value.find((item) => defaultCultureCodes.includes(item.languageCode))?.value ?? '';
  }
  return '';
};

export const getMetaFields = (item: DialogByIdFieldsFragment, isSeenByEndUser: boolean) => {
  const nOtherSeen = item.seenSinceLastUpdate?.filter((seenLogEntry) => !seenLogEntry.isCurrentEndUser).length ?? 0;
  const metaFields: InboxItemMetaField[] = [];

  if (isSeenByEndUser && nOtherSeen) {
    metaFields.push({
      type: 'seenBy',
      label: `${i18n.t('word.seenBy')} ${i18n.t('word.you')} ${i18n.t('word.and')} ${nOtherSeen} ${i18n.t('word.others')}`,
      options: {
        tooltip: item.seenSinceLastUpdate.map((seenLogEntry) => seenLogEntry.seenBy.actorName).join('\n'),
      },
    });
  } else if (nOtherSeen) {
    metaFields.push({
      type: 'seenBy',
      label: `${i18n.t('word.seenBy')} ${nOtherSeen} ${i18n.t('word.others')}`,
      options: {
        tooltip: item.seenSinceLastUpdate.map((seenLogEntry) => seenLogEntry.seenBy.actorName).join('\n'),
      },
    });
  } else if (isSeenByEndUser) {
    metaFields.push({
      type: 'seenBy',
      label: `${i18n.t('word.seenBy')} ${i18n.t('word.you')}`,
      options: {
        tooltip: item.seenSinceLastUpdate.map((seenLogEntry) => seenLogEntry.seenBy.actorName).join('\n'),
      },
    });
  }
  return metaFields;
};

const getMainContentReference = (
  args: { value: ValueType; mediaType: string } | undefined | null,
): MainContentReference | undefined => {
  if (typeof args === 'undefined' || args === null) return undefined;

  const { value, mediaType } = args;
  const url = getPropertyByCultureCode(value);

  /* TODO: add support for frontchannelembed+json;type=html */
  switch (mediaType) {
    case 'text/markdown':
    case 'application/vnd.dialogporten.frontchannelembed+json;type=markdown':
      return { url, mediaType: 'markdown' };
    default:
      return { url, mediaType: 'unknown' };
  }
};

export function mapDialogDtoToInboxItem(
  item: DialogByIdFieldsFragment | null | undefined,
  parties: PartyFieldsFragment[],
): DialogByIdDetails | undefined {
  if (!item) {
    return undefined;
  }
  const titleObj = item?.content?.title?.value;
  const additionalInfoObj = item?.content?.additionalInfo?.value;
  const summaryObj = item?.content?.summary?.value;
  const mainContentReference = item?.content?.mainContentReference;
  const endUserParty = parties?.find((party) => party.isCurrentEndUser);
  const dialogReceiverParty = parties?.find((party) => party.party === item.party);
  const actualReceiverParty = dialogReceiverParty ?? endUserParty;
  const serviceOwner = getOrganisation(item.org, 'nb');
  const isSeenByEndUser = item.seenSinceLastUpdate.find((seenLogEntry) => seenLogEntry.isCurrentEndUser) !== undefined;

  return {
    title: getPropertyByCultureCode(titleObj),
    summary: getPropertyByCultureCode(summaryObj),
    sender: {
      name: serviceOwner?.name ?? '',
      isCompany: true,
      imageURL: serviceOwner?.logo,
    },
    receiver: {
      name: actualReceiverParty?.name ?? '',
      isCompany: actualReceiverParty?.partyType === 'Organisation',
    },
    metaFields: getMetaFields(item, isSeenByEndUser),
    additionalInfo: getPropertyByCultureCode(additionalInfoObj),
    guiActions: item.guiActions.map((guiAction) => ({
      id: guiAction.id,
      url: guiAction.url,
      hidden: !guiAction.isAuthorized,
      priority: guiAction.priority,
      httpMethod: guiAction.httpMethod,
      title: getPropertyByCultureCode(guiAction.title),
      prompt: getPropertyByCultureCode(guiAction.prompt),
      isDeleteAction: guiAction.isDeleteDialogAction,
      disabled: !guiAction.isAuthorized,
    })),
    attachments: item.attachments.filter((attachment) => attachment.urls.length > 0),
    mainContentReference: getMainContentReference(mainContentReference),
    dialogToken: item.dialogToken!,
    activities: item.activities
      .map((activity) => ({
        id: activity.id,
        type: activity.type,
        createdAt: activity.createdAt,
        performedBy: activity.performedBy,
        description: getPropertyByCultureCode(activity.description),
      }))
      .reverse(),
    createdAt: item.createdAt,
    updatedAt: item.updatedAt,
  };
}
export const useDialogById = (parties: PartyFieldsFragment[], id?: string): UseDialogByIdOutput => {
  const partyURIs = parties.map((party) => party.party);
  const { data, isSuccess, isLoading } = useQuery<GetDialogByIdQuery>({
    queryKey: [QUERY_KEYS.DIALOG_BY_ID, id],
    staleTime: 1000 * 60 * 10,
    retry: 3,
    queryFn: () => getDialogsById(id!),
    enabled: typeof id !== 'undefined' && partyURIs.length > 0,
  });
  return {
    isLoading,
    isSuccess,
    dialog: mapDialogDtoToInboxItem(data?.dialogById.dialog, parties),
  };
};
