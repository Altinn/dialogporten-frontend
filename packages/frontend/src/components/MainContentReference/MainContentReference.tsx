import { Markdown } from 'embeddable-markdown-html';
import { useQuery } from 'react-query';
import type { DialogByIdDetails } from '../../api/useDialogById.tsx';
import styles from './mainContentReference.module.css';

export const MainContentReference = ({
  args,
  dialogToken,
}: { args: DialogByIdDetails['mainContentReference']; dialogToken: string }) => {
  const { data, isSuccess, error } = useQuery({
    queryKey: ['mainContentReference', args?.url],
    queryFn: () =>
      fetch(args!.url, {
        headers: {
          'Content-Type': 'text/plain',
          Authorization: `Bearer ${dialogToken}`,
        },
      }).then((res) => res.text()),
    enabled: args?.url !== undefined && args?.mediaType === 'markdown',
  });

  if (!args) {
    return null;
  }

  if (typeof error === 'string') {
    return <div data-id="dialog-main-content-reference-error">Error parsing 'mainContentReference': {error}</div>;
  }

  if (typeof data === 'string' && isSuccess) {
    return (
      <section data-id="dialog-main-content-reference" className={styles.mainContentReference}>
        <Markdown>{data}</Markdown>
      </section>
    );
  }

  return null;
};
