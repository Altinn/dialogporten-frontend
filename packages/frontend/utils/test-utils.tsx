import { type RenderOptions, render } from '@testing-library/react';
import type { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { SelectedDialogsContainer } from '..';
import { FeatureFlagProvider, featureFlags } from '../src/featureFlags';
import '../src/i18n/config.ts';
import { MockAuthProvider } from '../src/components/Login/MockAuthContext.tsx';

interface IExtendedRenderOptions extends RenderOptions {
  initialEntries?: string[];
}

export const createCustomWrapper = (
  ownQueryClient?: QueryClient,
  options?: Omit<IExtendedRenderOptions, 'wrapper'>,
) => {
  const queryClient =
    ownQueryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
        },
      },
    });

  return ({ children }: { children: React.ReactNode }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <FeatureFlagProvider flags={featureFlags}>
          <MockAuthProvider>
            <MemoryRouter initialEntries={options?.initialEntries ?? ['/']}>
              <SelectedDialogsContainer>{children}</SelectedDialogsContainer>
            </MemoryRouter>
          </MockAuthProvider>
        </FeatureFlagProvider>
      </QueryClientProvider>
    );
  };
};

export const customRender = (ui: ReactElement, options?: Omit<IExtendedRenderOptions, 'wrapper'>) => {
  const Wrapper = createCustomWrapper(undefined, options);

  return render(ui, { wrapper: Wrapper, ...options });
};
