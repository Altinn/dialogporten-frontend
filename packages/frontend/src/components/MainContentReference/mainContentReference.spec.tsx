import * as ReactQuery from '@tanstack/react-query';
import { QueryClient } from '@tanstack/react-query';
import { type RenderOptions, waitFor } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { createCustomWrapper, customRender } from '../../../utils/test-utils.tsx';
import { EmbeddableMediaType } from '../../api/useDialogById.tsx';
import { MainContentReference } from './MainContentReference.tsx';

const queryClient = new QueryClient();
const wrapper = createCustomWrapper(queryClient);
const mockDialogToken = 'mock-token';

vi.mock('@tanstack/react-query', async () => {
  const actual = await vi.importActual('@tanstack/react-query');
  return {
    ...actual,
    useQuery: vi.fn(),
  };
});

describe('MainContentReference Component', () => {
  it('should render markdown content', async () => {
    const mockContent = {
      url: 'https://altinn.mock/content',
      mediaType: EmbeddableMediaType.markdown,
    };

    const mockResponse = '# header ## subheader ### subsubheader';

    vi.spyOn(ReactQuery, 'useQuery').mockImplementation(
      vi.fn().mockReturnValue({ data: mockResponse, isLoading: false, isSuccess: true, isError: false }),
    );

    const { asFragment } = await waitFor(() =>
      customRender(<MainContentReference content={mockContent} dialogToken={mockDialogToken} />, {
        wrapper,
      } as RenderOptions),
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render html content', async () => {
    const mockContent = {
      url: 'https://altinn.mock/content',
      mediaType: EmbeddableMediaType.html,
    };

    const mockResponse = '<html><body><h1>header 1</h1></body></html>';

    vi.spyOn(ReactQuery, 'useQuery').mockImplementation(
      vi.fn().mockReturnValue({ data: mockResponse, isLoading: false, isSuccess: true, isError: false }),
    );

    const { asFragment } = await waitFor(() =>
      customRender(<MainContentReference content={mockContent} dialogToken={mockDialogToken} />, {
        wrapper,
      } as RenderOptions),
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
