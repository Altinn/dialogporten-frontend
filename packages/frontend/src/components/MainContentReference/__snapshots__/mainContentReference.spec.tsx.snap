// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainContentReference Component > should render html content 1`] = `
<DocumentFragment>
  <div
    class="_typography_cr720_1 mainContentReference"
    data-size="md"
  >
    <h1
      class="__fce-header-1"
    >
      header 1
    </h1>
  </div>
</DocumentFragment>
`;

exports[`MainContentReference Component > should render markdown content 1`] = `
<DocumentFragment>
  <div
    class="_typography_cr720_1 mainContentReference"
    data-size="md"
  >
    <h1
      class="__fce-header-1"
    >
      header ## subheader ### subsubheader
    </h1>
  </div>
</DocumentFragment>
`;
